crates/jiff-diesel/src/nullable.rs

```diff
@@ -0,0 +1,272 @@
+    ) -> diesel::deserialize::Result<Self> {
+        FromSql::from_nullable_sql(bytes).map(NullableDate)
+    }
+}
+
+/// A wrapper type for [`Option<jiff::civil::Time>`], can be used in `Queryable` derivations.
+#[derive(Clone, Copy, Debug, diesel::deserialize::FromSqlRow)]
+pub struct NullableTime(Option<crate::Time>);
+
+impl ToDiesel for Option<jiff::civil::Time> {
+    type Target = NullableTime;
+
+    fn to_diesel(self) -> NullableTime {
+        NullableTime(self.map(ToDiesel::to_diesel))
+    }
+}
+
+impl From<Option<jiff::civil::Time>> for NullableTime {
+    fn from(x: Option<jiff::civil::Time>) -> Self {
+        Self(x.map(Into::into))
+    }
+}
+
+impl From<NullableTime> for Option<jiff::civil::Time> {
+    fn from(x: NullableTime) -> Self {
+        x.0.map(Into::into)
+    }
+}
+
+impl<DB: Backend, ST> ToSql<ST, DB> for NullableTime
+where
+    Option<crate::Time>: ToSql<ST, DB>,
+{
+    fn to_sql<'b>(
+        &'b self,
+        out: &mut Output<'b, '_, DB>,
+    ) -> diesel::serialize::Result {
+        self.0.to_sql(out)
+    }
+}
+
+impl<DB: Backend, ST> FromSql<ST, DB> for NullableTime
+where
+    Option<crate::Time>: FromSql<ST, DB>,
+{
+    fn from_sql(
+        bytes: <DB as Backend>::RawValue<'_>,
+    ) -> diesel::deserialize::Result<Self> {
+        FromSql::from_sql(bytes).map(NullableTime)
+    }
+
+    fn from_nullable_sql(
+        bytes: Option<<DB as Backend>::RawValue<'_>>,
+    ) -> diesel::deserialize::Result<Self> {
+        FromSql::from_nullable_sql(bytes).map(NullableTime)
+    }
+}
+
+/// A wrapper type for [`Option<jiff::Span>`], can be used in `Queryable` derivations.
+#[derive(Clone, Copy, Debug, diesel::deserialize::FromSqlRow)]
+pub struct NullableSpan(Option<crate::Span>);
+
+impl ToDiesel for Option<jiff::Span> {
+    type Target = NullableSpan;
+
+    fn to_diesel(self) -> NullableSpan {
+        NullableSpan(self.map(ToDiesel::to_diesel))
+    }
+}
+
+impl From<Option<jiff::Span>> for NullableSpan {
+    fn from(x: Option<jiff::Span>) -> Self {
+        Self(x.map(Into::into))
+    }
+}
+
+impl From<NullableSpan> for Option<jiff::Span> {
+    fn from(x: NullableSpan) -> Self {
+        x.0.map(Into::into)
+    }
+}
+
+impl<DB: Backend, ST> ToSql<ST, DB> for NullableSpan
+where
+    Option<crate::Span>: ToSql<ST, DB>,
+{
+    fn to_sql<'b>(
+        &'b self,
+        out: &mut Output<'b, '_, DB>,
+    ) -> diesel::serialize::Result {
+        self.0.to_sql(out)
+    }
+}
+
+impl<DB: Backend, ST> FromSql<ST, DB> for NullableSpan
+where
+    Option<crate::Span>: FromSql<ST, DB>,
+{
+    fn from_sql(
+        bytes: <DB as Backend>::RawValue<'_>,
+    ) -> diesel::deserialize::Result<Self> {
+        FromSql::from_sql(bytes).map(NullableSpan)
+    }
+
+    fn from_nullable_sql(
+        bytes: Option<<DB as Backend>::RawValue<'_>>,
+    ) -> diesel::deserialize::Result<Self> {
+        FromSql::from_nullable_sql(bytes).map(NullableSpan)
+    }
+}
```

Comment:
I think these all make sense to me. I do somewhat wonder if it would be nicer to put these into a sub-module named `nullable`, and then remove `Nullable` from the type names. But I think perhaps it's not worth it if we only have two copies of each type.